libnetwork/drivers/bridge/bridge_test.go

```diff
@@ -1087,18 +1087,20 @@ func TestCreateParallel(t *testing.T) {
 		t.Fatalf("Failed to setup driver config: %v", err)
 	}
 
+	ipV4Data := getIPv4Data(t, "docker0")
```

Comment:
Not sure I understand the race.
Can you explain this more?

Reply:
I'm not 100% confident in this explanation. `getIPv4Data` is implemented in terms of [`netutils.ElectInterfaceAddresses`](https://pkg.go.dev/github.com/docker/docker/libnetwork/netutils#ElectInterfaceAddresses), which either returns all addresses for the named interface or, if the interface does not exist, returns an arbitrary address which does not conflict with any assigned addresses. In the old world where the tests would all run in the container's network namespace, the `docker0` interface would exist and therefore every call to `getIPv4Data("docker0")` would return the same set of addresses. There is no interface named `docker0` in a freshly-created network namespace, so it would return different sets of addresses once the previous sets are assigned to interfaces. Given that the test asserts exactly one network is created, it is expecting the other 99 attempts to conflict with the successful one. But when the `docker0` interface does not exist, the next call to `getIPv4Data` after a bridge network is created (and its addresses assigned to a network interface) would return a new, non-conflicting address range. If by chance some of the racing goroutines are randomly scheduled sequentially, more than one bridge network with non-conflicting address ranges could successfully be created, failing the test.